compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BUnionType.java

```diff
@@ -55,6 +55,7 @@
     public Boolean isAnyData = null;
     public Boolean isPureType = null;
     public boolean isCyclic = false;
+    public Boolean isAllErrorType = null;
```

Comment:
Do we really need this? I think we can live with the existing check.

Reply:
But without this, things get iterated again, isn't it?

Reply:
https://github.com/ballerina-platform/ballerina-lang/pull/35166/files#diff-51851f1451506fedd7c81b67a92282fada4eb00fe4728c70445d7262aed401f3R2400

Reply:
I meant to say that that's OK. 

There are quite a few optimizations we can do by caching type information, which we don't do atm. I'm guessing the approach we will take will mostly likely not depend on variables like this. IMO, we can live with iteration for now.

Reply:
This also opens up room for error given that we can add members to a union after creation - https://github.com/ballerina-platform/ballerina-lang/blob/16da1ac806f08c939b27a3b9f73ea4a11ef7e171/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BUnionType.java#L252. In which case `isAllErrorType` has to be recomputed.

Reply:
Ah, that's a good point. will this attribute.